python/pyiceberg/catalog/hive.py

```diff
@@ -229,24 +290,43 @@ def create_table(
             ValueError: If the identifier is invalid
         """
         database_name, table_name = self.identifier_to_database_and_table(identifier)
-        current_time_millis = int(time.time())
+        seconds_since_epoch = int(time.time())
+
+        location = self._resolve_table_location(location, database_name, table_name)
+
+        metadata_location = f"{location}metadata/{uuid.uuid4()}.metadata.json"
+        metadata = TableMetadataV2(
+            location=location,
+            schemas=[schema],
```

Comment:
The schema field IDs and partition field IDs need to be reassigned to ensure that they are consistent because we don't trust that users pass them in correctly. Passing them directly is okay for this PR, but we should not release until we have reassignment done.

Reply:
Good one, we do have validators on them when we initialize the Pydantic models, but there is no re-assignment being done. Do you want to set them to zero? We could also set the `schema_id` by default to zero?

Reply:
Here's what we do in Java: https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableMetadata.java#L88-L133

When the table metadata builder adds a schema, spec, or order, the builder will check whether it is an existing order and find out the ID to assign. We should do something similar here.

This probably isn't something to do in this PR (since this is focused on Hive) but introducing the builder would probably be a good idea. That's how we also accumulate the change set that we pass to the REST catalog when committing table changes. I'd say let's move forward with what you have here (except passing the IDs rather than hard-coding to 0) and we can add reassignment later.

Reply:
Thanks for pointing out the Java code. I like it and agree that it is best to do it in separate PR.